istioctl/pkg/writer/pilot/status.go

```diff
@@ -117,3 +138,86 @@ func xdsStatus(sent, acked string) string {
 	// Since the Nonce changes to uuid, so there is no more any time diff info
 	return "STALE"
 }
+
+// PrintAll takes a slice of Istiod syncz responses and outputs them using a tabwriter
+func (s *XdsStatusWriter) PrintAll(statuses map[string]*xdsapi.DiscoveryResponse) error {
+	w, fullStatus, err := s.setupStatusPrint(statuses)
+	if err != nil {
+		return err
+	}
+	for _, status := range fullStatus {
+		if err := xdsStatusPrintln(w, status); err != nil {
+			return err
+		}
+	}
+	return w.Flush()
+}
+
+func (s *XdsStatusWriter) setupStatusPrint(drs map[string]*xdsapi.DiscoveryResponse) (*tabwriter.Writer, []*xdsWriterStatus, error) {
+	// Gather the statuses before printing so they may be sorted
+	var fullStatus []*xdsWriterStatus
+	for _, dr := range drs {
+		for _, resource := range dr.Resources {
+			switch resource.TypeUrl {
+			case "type.googleapis.com/envoy.service.status.v3.ClientConfig":
+				clientConfig := xdsstatus.ClientConfig{}
+				err := ptypes.UnmarshalAny(resource, &clientConfig)
+				if err != nil {
+					return nil, nil, fmt.Errorf("could not unmarshal ClientConfig: %w", err)
+				}
+				cds, lds, eds, rds := getSyncStatus(clientConfig.GetXdsConfig())
+				cp := multixds.CpInfo(dr)
+				fullStatus = append(fullStatus, &xdsWriterStatus{
+					proxyID:        clientConfig.GetNode().GetId(),
+					istiodID:       cp.ID,
+					istiodVersion:  cp.Info.Version,
+					clusterStatus:  cds,
+					listenerStatus: lds,
+					routeStatus:    rds,
+					endpointStatus: eds,
+				})
+			default:
+				return nil, nil, fmt.Errorf("/debug/syncz unexpected resource type %q", resource.TypeUrl)
+			}
+		}
+	}
+	if len(fullStatus) == 0 {
+		return nil, nil, fmt.Errorf("no proxies found (checked %d istiods)", len(drs))
+	}
+
+	w := new(tabwriter.Writer).Init(s.Writer, 0, 8, 5, ' ', 0)
+	_, _ = fmt.Fprintln(w, "NAME\tCDS\tLDS\tEDS\tRDS\tISTIOD\tVERSION")
+
+	sort.Slice(fullStatus, func(i, j int) bool {
+		return fullStatus[i].proxyID < fullStatus[j].proxyID
+	})
+	return w, fullStatus, nil
+}
+
+func xdsStatusPrintln(w io.Writer, status *xdsWriterStatus) error {
+	_, _ = fmt.Fprintf(w, "%v\t%v\t%v\t%v\t%v\t%v\t%v\n",
```

Comment:
Better to return `err` if the formating fails?

Reply:
My practice is to ignore failing Printf because if it fails there will be nowhere to print the error later.  I will give it a shot though.